Enable scrollbar-color by default

nmoucht opened the pull request:
#### 102e55a24491d695148eb5453a5607b66bb8602b
<pre>
Enable scrollbar-color by default
<a href="https://bugs.webkit.org/show_bug.cgi?id=297481">https://bugs.webkit.org/show_bug.cgi?id=297481</a>
<a href="https://rdar.apple.com/158436083">rdar://158436083</a>

Reviewed by Aditya Keerthi.

Enable scrollbar-color by default on platforms that support it.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultScrollbarColorEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:

Canonical link: <a href="https://commits.webkit.org/298827@main">https://commits.webkit.org/298827@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8255103467803460b6f420f025e2eeaf9a3357c2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116803 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36467 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27040 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122879 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/67391 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b27a9158-4727-45b5-be46-f006b7dc3bf7) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37165 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45068 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88697 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/43109 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/929e8a53-db44-44dc-94e4-4b3266ccc990) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119752 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29629 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104777 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69166 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28692 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22882 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66544 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/108917 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99008 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23037 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126013 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/115331 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43702 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32823 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97365 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44066 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100978 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97163 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24735 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42489 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20429 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39710 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43588 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/49184 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/144029 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43055 "Built successfully") | | [💥 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37084 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46394 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44760 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -1316,16 +1316,19 @@ CSSScrollAnchoringEnabled:
   category: css
   humanReadableName: "CSS scrollbar-color property"
   humanReadableDescription: "Enable scrollbar-color CSS property"
+  condition:
```
why is this empty?

nmoucht replied:
Removed.

pxlcoder commented on Source/WebCore/platform/ScrollTypes.cpp:
```diff
@@ -216,4 +216,13 @@ TextStream& operator<<(TextStream& ts, ScrollSnapPointSelectionMethod option)
     return ts;
 }
 
+bool defaultScrollbarColoredEnabled()
```
This should be declared and implemented in `WebPreferencesDefaultValues.h`/`.cpp` in WebKit.

nmoucht replied:
Fixed.

pxlcoder commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -1316,17 +1316,17 @@ CSSScrollAnchoringEnabled:
     WebKitLegacy:
       default: false
     WebKit:
-      default: false
```
This should then just be `default: defaultScrollbarColoredEnabled()` in WebKit.

nmoucht replied:
Fixed.

pxlcoder commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -1316,17 +1316,17 @@ CSSScrollAnchoringEnabled:
-      default: false
+      default: true
     WebCore:
-      default: false
```
WebCore should be `false` by default. The value would propagate from WebKit.

nmoucht replied:
Fixed.

pxlcoder commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -1316,15 +1316,15 @@ CSSScrollAnchoringEnabled:
       default: false
     WebKit:
-      default: false
+      default: WebKit::defaultScrollbarColoredEnabled()
```
I don't think the `WebKit::` prefix is necessary.

nt1m commented on Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
```diff
@@ -429,4 +429,13 @@ bool defaultIOSurfaceLosslessCompressionEnabled()
 }
 #endif
 
+bool defaultScrollbarColoredEnabled()
```
```suggestion
bool defaultScrollbarColorEnabled()
```

nmoucht committed:
Enable scrollbar-color by default
https://bugs.webkit.org/show_bug.cgi?id=297481
rdar://158436083

Reviewed by Aditya Keerthi.

Enable scrollbar-color by default on platforms that support it.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultScrollbarColorEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:

Canonical link: https://commits.webkit.org/298827@main
